### PR TITLE
[SPARK-41894][SS][TESTS] Restore the write permission of `commitDir` after run `testAsyncWriteErrorsPermissionsIssue`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims to restore the write permission of `commitDir` after run `testAsyncWriteErrorsPermissionsIssue` in `AsyncProgressTrackingMicroBatchExecutionSuite` so that `mvn clean` can run successfully.

### Why are the changes needed?
Make `mvn clean` can run successfully after run `mvn test` with `AsyncProgressTrackingMicroBatchExecutionSuite`

### Does this PR introduce _any_ user-facing change?
No, just for test

### How was this patch tested?

- Pass Github Actions
- Manual test

```
build/mvn clean install -pl sql/core -am -DskipTests
build/mvn clean test -pl sql/core -Dtest=none -DwildcardSuites=org.apache.spark.sql.execution.streaming.AsyncProgressTrackingMicroBatchExecutionSuite
build/mvn clean -pl sql/core
```

**Before**

`build/mvn clean -pl sql/core` run failed

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-clean-plugin:3.1.0:clean (default-clean) on project spark-sql_2.12: Failed to clean project: Failed to delete /${basedir}/sql/core/target/tmp/streaming.metadata-4d41fbcc-d517-4159-961c-95688dadd2c8/offsets/0 -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

**After**

All three commands run successfully.

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @HyukjinKwon @jerrypeng

File: sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/AsyncProgressTrackingMicroBatchExecutionSuite.scala [hunk 1]
```diff
@@ -750,6 +750,12 @@ class AsyncProgressTrackingMicroBatchExecutionSuite
             }
             e.getCause.getCause.getMessage should include("Permission denied")
           }
+      },
+      Execute { _ =>
+        // SPARK-41894: Restore the write permission of `commitDir`
+        // so that `mvn clean` can run successfully.
+        val commitDir = new File(checkpointLocation + path)
+        commitDir.setWritable(true)
```
Comment (LuciferYang) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/AsyncProgressTrackingMicroBatchExecutionSuite.scala:758]: https://github.com/apache/spark/blob/58d5dc3d573dfbb6d21ea41d101550146756f45b/sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/AsyncProgressTrackingMicroBatchExecutionSuite.scala#L736-L743

Line 741 `commitDir.setReadOnly()`

Comment (LuciferYang) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/AsyncProgressTrackingMicroBatchExecutionSuite.scala:758]: Is this ok? Or put the new action to finally block?

**Comment by HyukjinKwon** [comment]
cc @HeartSaVioR

File: sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/AsyncProgressTrackingMicroBatchExecutionSuite.scala (hunk 1 above)
Comment (HeartSaVioR) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/AsyncProgressTrackingMicroBatchExecutionSuite.scala:758]: I'd prefer restoring the permission in finally block.

Comment (LuciferYang) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/AsyncProgressTrackingMicroBatchExecutionSuite.scala]: [7716abc](https://github.com/apache/spark/pull/39406/commits/7716abc32aa3b070e52b1e0f087a5fcc089c0ac3) change to use `finally block`

**Comment by LuciferYang** [comment]
friendly ping @HyukjinKwon @HeartSaVioR

**Review by HeartSaVioR — APPROVED** [review]
+1

**Comment by HeartSaVioR** [comment]
Thanks! Merging to master.

**Comment by LuciferYang** [comment]
Thanks @HeartSaVioR

**Pull request closed without merging**